designer_v2/lib/router.dart

```diff
@@ -1,82 +1,152 @@
-import 'package:designer_v2/pages/error_page.dart';
-import 'package:designer_v2/pages/login_page.dart';
-import 'package:designer_v2/pages/splash_page.dart';
-import 'package:designer_v2/pages/study_dashboard_screen.dart';
-import 'package:designer_v2/services/app_service.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:go_router/go_router.dart';
+import 'package:studyu_designer_v2/features/app_controller.dart';
+import 'package:studyu_designer_v2/features/dashboard/dashboard_page.dart';
+import 'package:studyu_designer_v2/common_views/pages/error_page.dart';
+import 'package:studyu_designer_v2/features/auth/login_page.dart';
+import 'package:studyu_designer_v2/common_views/pages/splash_page.dart';
+import 'package:studyu_designer_v2/repositories/auth_repository.dart';
+import 'package:studyu_designer_v2/utils/combined_stream_notifier.dart';
+import 'package:studyu_designer_v2/utils/debug_print.dart';
 
-// List of all pages in the application
-// To create a new page, 1) add an entry to the enum and 2) add the page to the AppRouter class
+/// To create a new page:
+/// 1) add an entry to the [RouterPage] enum
+/// 2) add the page widget to the [routes] list
+
+/// List of all pages in the application
 enum RouterPage {
-  dashboard(title: "dashboard", path: "/"),
-  splash(title: "splash", path: "/splash"),
-  login(title: "login", path: "/login"),
-  error(title: "error", path: "/error");
+  dashboard(id: "dashboard", path: "/studies"),
+  dashboardOwned(id: "dashboardOwned", path: "owned"),
+  dashboardShared(id: "dashboardShared", path: "shared"),
+  studyRegistry(id: "studyRegistry", path: "registry"),
+  splash(id: "splash", path: "/splash"),
+  login(id: "login", path: "/login"),
+  error(id: "error", path: "/error");
```

Comment:
This can become pretty messy with time. Maybe we can find a way to structure the paths more hierarchically?

Reply:
Agreed, but not sure how to do this if we want to keep using an Enum here. I would not worry about it for now until it becomes a problem.